docs/development/pigs/pig-029.rst

```diff
@@ -0,0 +1,207 @@
+********************************
+PIG 29 - SkyModel Restructuring
+********************************
+
+* Author: Atreyee Sinha, Maxime Regard
+* Created: ``2024-12-13``
+* Accepted: ``xx``
+* Status: xx
+* Discussion: `#4381`_
+
+Abstract
+========
+
+This PIG intends to resturcture `gammapy.modelling.models` to allow for flexible model definitions on arbitrary variables.
+
+Motivation
+==========
+
+Currently, gammapy supports Temporal, Spectral and Spatial models, which are bundled together in the `SkyModel` class for modelling and fitting. Despite the underlying `maps` framework being completely flexible, the modelling framework follows a fixed API allowing only spectral (compulsory), temporal and spatial models. If a user wants to model the behaviour in other parameters, say phase or frequency, it requires expert knowledge and is often performed using other fitting packages, eg: see https://github.com/cgalelli/gammapy_SyLC/tree/main/gammapy_SyLC.
+
+Currently, the `MapDataset` and the `FluxPointsDataset` support additional axes, so the main changes are required in the `MapEvaluator` and in the model definitions
+
+
+Use cases
+=========
+
+1. Fitting other parameters like phase, frequency, etc
+2. Reduction of code duplication: Analytic shapes like the power law is defined separately for `PowerLawSpectralModel`, `PowerLawTeLmporalModel` and `PowerLawNormSpectralModel`, which differ only in the unit handling
+3. Separation of units and values: This is a side benefit which can be useful if move to using JAX arrays in the future.
+
+
+Proposed API
+============
+
+We propose to have a general unit-less analytic class definition, inheriting from `ModelBase` The unit handling responsibility lies on the user.
+Internal support and checks and be provided for common input parameters (eg: time/energy) and the expected output units. 
+
+So, we move from 
+
+`SkyModel(spectral_model=, temporal_model=)`
+to 
+
+`SkyModelGeneral(submodels={"energy_true": p, "time":t})`
```

Comment:
what happens to the spatial model ?

Reply:
What we do for the 2D models is a question left open. One can say, eg, ` submodels={spatial: GaussianSpatialModel}` if we dont want to change the spatial models.

2D general models maybe useful, eg in the camera plane instead of on the sky. But I dont know how practically useful it can be